### DISCUSSION TITLE
Is there a way to update the style of a single annotation (redraw the annotation)?

### DISCUSSION THREAD

**Discussion by flamsens** [annotorious/annotorious#499] [Q&A] [answered]
I'm looking for a way to update the style of an annotation (redraw an annotation). Now it is possible to write your own style function  and use some state known outside of Annotorious, but I don't know how to redraw/update one or multiple annotations when that state changes. 

There is an `updateAnnotation` method, but that is to update annotaton data, not to refresh the render style of an annotation. It would be nice to trigger a redraw somehow.

**Comment by rsimon** [comment, marked as answer]
If you replace the style function, that would trigger a redraw. Or you could remove and re-add the annotation if you are worried about redraw performance.

**Reply by flamsens** [reply]
Replacing the style functions works fine, thank you! Maybe a `redraw` method with a line in the documentation could be usefull? Performance is excellent by the way, even with 100+ annotations.

**Reply by rsimon** [reply]
I was thinking about some kind of "annotation state" method at some point. (I.e. something `anno.setState(annotationId, {...custom}`), and then this could trigger a repaint for a specific annotation, and the custom state would get passed along to the style function, together with the normal annotation state. But then I wasn't sure this would be too awkward. Maybe it would make sense after all...

**Reply by rsimon** [reply]
BTW: performance should be good with 1.000s of annotations. But it depends on boxes vs. polygons, polygon complexity, other stuff going on in the UI etc. If you have experiences with large number of annotations I'd really like to hear them. I haven't really load-tested under realistic conditions myself yet.

**Reply by flamsens** [reply]
A `setState` method would work for my usecase, but not only for custom states. I think one should be able to change the 'normal' annotation states as well. 

I'm working on a cuneiform research project where I display sign annotations on a clay tablet and a transliteration next to the image. When I hover a sign on the image, I highlight the corresponding token in the transliteration. When I hover a token in the transliteration, I highlight the corresponding sign on the image. But words are made of multiple signs in Akkadian. So if I hover a word in a transliteration/transcription, I want to highlight multiple signs on the image. And maybe if I hover a sign on the image, I want to highlight the other signs belonging to the same word on the image as well.

So here it would help to be able to change the (hover) state of annotations outside of Annotorious..

**Reply by rsimon** [reply]
That's exactly what the custom states would be there for. I.e. you'd introduce a "signHovered" state and let the style react to that. The built-in states would be reserved for actual hover/selection. Manipulating those would IMO just introduce ambiguity.
